libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -153,7 +153,7 @@ static DNNReturnType get_input_tf(void *model, DNNData *input, const char *input
     TF_DeleteStatus(status);
 
     // currently only NHWC is supported
-    av_assert0(dims[0] == 1);
+    av_assert0(dims[0] == 1 || dims[0] == -1);
```

Comment:
is this change enough to "support model with random input"?

btw, the title 'random input' is not clear.

Reply:
Just for fixing the error of "input dims[0] != 1". Maybe i should change this commit message to something like: dnn/dnn_backend_tf: fix error of model input batch_size = -1 ?

Reply:
dims[0] is batch size, we'd put all the changes relative to batch size together as a single patch

Reply:
OK